src/utils.rs

```diff
@@ -40,29 +41,25 @@ pub fn do_exec(path: impl AsRef<Path>, args: &[String], envs: &[String]) -> Resu
         .iter()
         .map(|s| CString::new(s.to_string()).unwrap_or_default())
         .collect();
-    let envs: Vec<CString> = envs
-        .iter()
-        .map(|s| CString::new(s.to_string()).unwrap_or_default())
-        .collect();
 
-    unsafe {
-        clearenv()?;
-    }
-    for e in envs {
-        putenv(&e)?
-    }
+    // clear env vars
+    env::vars().for_each(|(key, _value)| std::env::remove_var(key));
```

Comment:
I'm questioning myself why I was using libc. I must have been sleepy.

Reply:
Libc seemed obvious probably because you were already using it. Honestly the code was a little more concise too since libc takes "key=value" as one string and rust takes key and value as separate strings so I had to split on "=".